recipes/zeus_expected/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
+find_package(zeus_expected REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE zeus::expected)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
+if (MSVC)
+    target_compile_options(${PROJECT_NAME} PRIVATE "/Zc:__cplusplus")
+endif ()
```

Comment:
Why is it needed? What error does it prevent when running the test package?

Reply:
the library utilizes `__cplusplus` to check what standard is in use, without `/Zc:__cplusplus`, programs will not compile (fail with static_assert). 

A workaround is to make use of `_MSVC_LANG`. To make this library for wider and eaiser use, I am considering that approach and would make a change in the future.

Tell me if the current `/Zc:__cplusplus` way could have any harm to conan. Should this recipe wait until the library abandon the requirement of `/Zc:__cplusplus` ?

Reply:
But the `__cplusplus` should be evaluated by default in Visual Studio as well: https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170

We are passing `cxx_std_17`, it should be enough to work the test package.

Reply:
@X1aomu Okay! Now I see, so we need to move it to `package_info` instead. Otherwise, users that consume this package, will need to pass that flag manually. I'll point the change in the code.

Reply:
Well, as the author of zeus_expected, i just released a [new version](https://github.com/zeus-cpp/expected/releases/tag/v1.1.0) to drop the requirement of `/Zc:__cplusplus`.

I'd like to let the conan recipe start from that new version. Thus conan users would forget about `/Zc:__cplusplus` :)

Reply:
Awesome!!! 👏